docs/grammar/SolidityParser.g4

```diff
@@ -421,12 +430,14 @@ inlineArrayExpression: LBrack (expression ( Comma expression)* ) RBrack;
 /**
  * Besides regular non-keyword Identifiers, some keywords like 'from' and 'error' can also be used as identifiers.
  */
-identifier: Identifier | From | Error | Revert | Global;
+identifier: Identifier | From | Error | Revert | Global | Suffix;
 
 literal: stringLiteral | numberLiteral | booleanLiteral | hexStringLiteral | unicodeStringLiteral;
 
 literalWithSubDenomination: numberLiteral SubDenomination;
 
+suffixedLiteral: literal identifier;
```

Comment:
Actually, just realized that this is not entirely correct. It does not have to be a single literal. Can be member access as well.

Reply:
Not sure if I follow here. Should we be doing something like this? 

```suggestion
memberAcess: identifier (Period identifier)+;

suffixedLiteral: (literal | memberAcess) identifier;
```

What would be a member access that is equivalent to a literal?

Reply:
I mean that something like `123 M.suffix` is allowed because you can do `import "A.sol" as M` and `A.sol` can have `suffix` defined in it. And jsut `identifier` does not allow it.

Not sure if we want to define `memberAccess` explicitly though. I see that inside `expression` that's defined in place like this:
```
expression Period (identifier | Address) # MemberAccess
```

So I guess the solution is either to repeat that for `suffixedLiteral` or indeed define it as `memberAccess`.

Reply:
Ah, ok, understood it now. I was thinking about the `literal`, not the `identifier`.
We have a test with suffix function imported from other file:
https://github.com/ethereum/solidity/blob/65188cafc3a277c0dfa74fb09c17e16f99f19d31/test/libsolidity/syntaxTests/literalSuffixes/usableAsSuffix/imported_function_as_suffix.sol#L6-L14

I have removed the failing test in `chk_antlr_grammar` before running it locally and then got no errors, so I think that the current grammar already covers that somehow, but not sure.

Reply:
> I think that the current grammar already covers that somehow, but not sure.

But how? It can't be. `expression` does account for that explicitly so the literals should have to as well.